indico/modules/events/registration/__init__.py

```diff
@@ -77,7 +75,7 @@ def _inject_event_header(event, **kwargs):
                 .order_by(db.func.lower(RegistrationForm.title))
                 .all())
 
-    registrations = get_published_registrations(event)
+    registrations = event.published_registrations
```

Comment:
I meant in `events/registration/display/event_header.html` (you currently pass `event` to it, so in there you could access `event.published_registrations` directly)